File: commands/push-pull-commands/pull.js
import simpleGit from "simple-git";

const git = simpleGit();

export default {
  command: "pull",
  describe: "Pull changes from a remote repository",
  builder: () => {},
  handler: async () => {
    try {
      await git.pull("origin", "main");
      console.log("Changes pulled from a remote repository");
    } catch (error) {
      console.error("Error pulling changes:", error.message);
    }
  },
};

File: cmd/pull.go
package cmd

import (
	"os"
	"os/exec"

	"github.com/spf13/cobra"
)

var PullCmd = &cobra.Command{
	Use: "pull [url]",
	Short: "Pull changes from a remote repository",
	Run: func(cmd *cobra.Command, args []string) {
		if err := gitPull(); err != nil {
			cmd.Println("Error pulling changes: ", err)
			return
		}

		cmd.Println("Changes pulled successfully from remote repository")

	},
}

func gitPull() error {
	cmd := exec.Command("git", "pull")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	return cmd.Run()
}